20.11/basetask.sql

```diff
@@ -0,0 +1,154 @@
+create database BookLibrary;
+
+use BookLibrary;
+
+create table Authors(
+	Id uniqueidentifier not null primary key,
+	FirsName varchar(255),
+	LastName varchar(255),
+	Country varchar(255),
+	BirthDate date);
+
+create table Books(
+	Id uniqueidentifier not null primary key,
+	Name varchar(255),
+	AuthorId uniqueidentifier foreign key references Authors(Id)on delete cascade,
+	Year int);
+
+create table Users(
+	Id uniqueidentifier not null primary key,
+	FirsName varchar(255),
+	LastName varchar(255),
+	Email varchar(255),
+	BirthDate date,
+	Age int,
+	Address varchar,
+	ExpiredDate date);
+
+create table UserBooks(
+	Id uniqueidentifier not null primary key,
+	UserId uniqueidentifier foreign key references Users(Id)on delete cascade,
+	BookId uniqueidentifier foreign key references Books(Id)on delete cascade,
+	CreatedDate datetime);
+
+
+--�������� ������� ��� ������� UserBooks, ����� ��� ���������� ������ � ������� 
+--CreatedDate ��� ����������� ������� ������������ ������� ����. 
+--������� ������ �������� ����� ��� ��������, ����� ����������� ������ ��� 1 ������
+--���������: ����������� GETDATE().
+
+create trigger CreatedDate on UserBooks
+	for insert
+	as
+	update UserBooks 
+	set CreatedDate = GetDate()
```

Comment:
This will set CreateDate of ALL records to current date each time new record added. You have to set CreatedDate only to values that were added. You should use 'inserted' table.

https://learn.microsoft.com/en-us/sql/relational-databases/triggers/use-the-inserted-and-deleted-tables?view=sql-server-ver16